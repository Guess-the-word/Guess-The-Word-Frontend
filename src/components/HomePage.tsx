import React from "react";

export const HomePage = () => {
  return <h2>Welcome To Guess The Word</h2>;
};
