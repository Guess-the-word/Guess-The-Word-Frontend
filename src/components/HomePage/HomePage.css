.mainContainer {
    height: 100%;
}

.logoRow {
    height: 60%;
    background-color: rgb(246, 74, 74);
}

.inputRow {
    height: 40%;
    background-color: #ffd14d;
    text-align: center;
}

.logoImage {
    margin: auto;
}


/* .roomInput {
    width: 30% !important;
} */

.containerCheck {
    display: flex;
    align-items: center;
}