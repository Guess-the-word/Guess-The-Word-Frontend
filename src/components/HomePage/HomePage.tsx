import React, { useState } from "react";
import "./HomePage.css";
import Container from "react-bootstrap/Container";
import Row from "react-bootstrap/Row";
import Logo from "./simple_logo.svg";
import Image from "react-bootstrap/Image";
import Button from "react-bootstrap/Button";
import Form from "react-bootstrap/Form";
import Col from "react-bootstrap/Col";
import Alert from "react-bootstrap/Alert";
import axios from "axios";
import { Redirect } from "react-router-dom";

export const HomePage = (props: any) => {
    const [roomName, setRoomName] = useState("");
    const [showError, setShowError] = useState(false);
    const [errorMessage, setErrorMessage] = useState("");
    const [show, setShow] = useState(false);

    const validateInput = (inputSize: number) => {
        const letterNumber = /^[0-9a-zA-Z]+$/;
        if (inputSize < 4 || inputSize > 10) {
            setShowError(true);
            setShow(true);
            setErrorMessage("Length should be between 4 and 10 characters");
        } else if (!letterNumber.test(roomName)) {
            setShowError(true);
            setShow(true);
            setErrorMessage(
                "Room name should only include letters and numbers"
            );
        } else {
            setShowError(false);
        }
    };
    const submitRoomName = async (roomName: string) => {
        console.log(roomName);
        try {
            // const res = await axios.get("/room");
            // console.log(res);
            props.history.push({
                pathname: "/room",
                state: [roomName],
            });
        } catch (e) {
            console.log(e);
        }
    };

    function handleRoomNameChange(
        e: React.ChangeEvent<HTMLInputElement>
    ): void {
        setRoomName(e.target.value);
        validateInput(e.target.value.length);
    }

    return (
        <Container fluid className="mainContainer">
            <Row className="justify-content-center logoRow">
                <Image src={Logo} alt="service logo" className="logoImage" />
            </Row>
            <Row className="justify-content-center inputRow">
                <Container>
                    <div
                        style={{
                            height: "3rem",
                            width: "26rem",
                            margin: "1rem auto",
                        }}
                    >
                        {show && showError && (
                            <Alert variant="danger" style={{ padding: "10px" }}>
                                {errorMessage}
                            </Alert>
                        )}
                    </div>
                    <div>
                        <Form.Group>
                            <Col sm={{ span: 4, offset: 4 }}>
                                <Form.Control
                                    type="text"
                                    placeholder="Enter Room Name"
                                    className="mt-3"
                                    onChange={(
                                        e: React.ChangeEvent<HTMLInputElement>
                                    ) => handleRoomNameChange(e)}
                                />
                            </Col>
                        </Form.Group>
                        <Button
                            onClick={() => {
                                validateInput(roomName.length);
                                submitRoomName(roomName);
                            }}
                            variant="primary"
                        >
                            Submit
                        </Button>
                    </div>
                </Container>
            </Row>
        </Container>
    );
};
