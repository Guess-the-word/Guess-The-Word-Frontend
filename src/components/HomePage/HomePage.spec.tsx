import { render } from "@testing-library/react";
import React from "react";
import { HomePage } from "./HomePage";

const renderHomePage = () => {
  return render(<HomePage />);
};

describe("The HomePage Suite", () => {
  it("should match the snapshot", () => {
    const { container } = renderHomePage();
    expect(container).toMatchSnapshot();
  });
});
