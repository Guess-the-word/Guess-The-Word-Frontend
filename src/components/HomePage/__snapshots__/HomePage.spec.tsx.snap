// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The HomePage Suite should match the snapshot 1`] = `
<div>
  <div
    class="mainContainer container-fluid"
  >
    <div
      class="justify-content-center logoRow row"
    >
      <img
        alt="service logo"
        class="logoImage"
        src="simple_logo.svg"
      />
    </div>
    <div
      class="justify-content-center inputRow row"
    >
      <div
        class="container"
      >
        <div
          style="height: 3rem; width: 26rem; margin: 1rem auto;"
        />
        <div>
          <div
            class="form-group"
          >
            <div
              class="col-sm-4 offset-sm-4"
            >
              <input
                class="mt-3 form-control"
                placeholder="Enter Room Name"
                type="text"
              />
            </div>
          </div>
          <button
            class="btn btn-primary"
            type="button"
          >
            Submit
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
